app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterFragmentPresenterTest.kt

```diff
@@ -0,0 +1,112 @@
+import io.reactivex.internal.schedulers.ExecutorScheduler
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.schedulers.Schedulers
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.verify
+import org.mockito.MockitoAnnotations
+import java.util.concurrent.Executor
+import java.util.concurrent.TimeUnit
+
+const val ZERO_VALUE = 0
+const val TEN_VALUE = 10
+const val EMPTY_VALUE = ""
+
+class CharacterFragmentPresenterTest {
+
+    @Mock
+    private lateinit var view: CharacterFragmentView
+    @Mock
+    private lateinit var charactersFragment: CharacterFragment
+    @Mock
+    private lateinit var getSingleCharacterServiceUseCase: GetSingleCharacterServiceUseCase
+
+    private lateinit var model: CharacterFragmentModel
+    private lateinit var presenter: CharacterFragmentPresenter
+
+    companion object {
+
+        @BeforeClass
+        @JvmStatic
+        fun setUpClass() {
+            val immediate = object : Scheduler() {
+                internal var noDelay = ZERO_VALUE
+
+                override fun scheduleDirect(run: Runnable, delay: Long, unit: TimeUnit): Disposable {
+                    return super.scheduleDirect(run, noDelay.toLong(), unit) // Prevents StackOverflowErrors when scheduling with a delay
+                }
+
+                override fun createWorker(): Scheduler.Worker {
+                    return ExecutorScheduler.ExecutorWorker(Executor { it.run() })
+                }
+            }
+            // These avoid ExceptionInInitializerError when testing methods that contains RxJava Schedulers
+            RxJavaPlugins.setInitIoSchedulerHandler { scheduler -> immediate }
+            RxJavaPlugins.setInitComputationSchedulerHandler { scheduler -> immediate }
+            RxJavaPlugins.setInitNewThreadSchedulerHandler { scheduler -> immediate }
+            RxJavaPlugins.setInitSingleSchedulerHandler { scheduler -> immediate }
+            RxAndroidPlugins.setInitMainThreadSchedulerHandler { scheduler -> immediate }
+        }
+
+        private const val SIZE = TEN_VALUE
+    }
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.initMocks(this)
+        RxAndroidPlugins.setInitMainThreadSchedulerHandler { scheduler -> Schedulers.trampoline() }
+        model = CharacterFragmentModel(getSingleCharacterServiceUseCase)
+        presenter = CharacterFragmentPresenter(view, model)
+    }
+
+    @Test
+    fun serviceResponseWithError() {
+        `when`(getSingleCharacterServiceUseCase.invoke()).thenReturn(Single.error(Exception(EMPTY_VALUE)))
+        presenter.init(charactersFragment)
+        verify(getSingleCharacterServiceUseCase).invoke()
+        verify(view).showToastNetworkError(EMPTY_VALUE)
+        verify(view).hideLoading()
+    }
+
+    @Test
+    fun serviceResponseWithItemToShow() {
+        val comic = Comics(TEN_VALUE, TEN_VALUE, EMPTY_VALUE)
```

Comment:
remove these three first lines and put them in a separated method pls

Reply:
Done